ouroboros-consensus/src/Ouroboros/Consensus/Node/Run.hs

```diff
@@ -98,4 +111,16 @@ class ( LedgerSupportsProtocol           blk
       , ShowProxy                (Header blk)
       , ShowProxy                 (Query blk)
       , ShowProxy           (TxId (GenTx blk))
-      ) => RunNode blk
+      ) => RunNode blk where
+  -- | A protocol may or may not be able to determine from the 'Header' whether
+  -- the header's block was self-issued.
+  --
+  -- TODO Only the @Shelley@ currently returns anything other than
```

Comment:
We'll open an Issue to support Byron too, since at least on mainnet, no one is forging Byron blocks.

Reply:
I opened Issue 2936 (see link below).